tools/minify/README.md

```diff
@@ -0,0 +1,171 @@
+Telegraf configuration file(s). Assuming your configuration is
+in `/etc/telegraf/telegraf.conf` you can run
+
+```shell
+# ./tools/minify/minify --config /etc/telegraf/telegraf.conf
+```
+
+to build a Telegraf binary tailored to your configuration.
+You can also specify a configuration directory similar to
+Telegraf itself. To additionally use the configurations in
+`/etc/telegraf/telegraf.d` run
+
+```shell
+# ./tools/minify/minify         \
+    --config     /etc/telegraf/telegraf.conf \
+    --config-dir /etc/telegraf/telegraf.d
+```
+
+Configurations can also be retrieved from remote locations just
+like for Telegraf.
+
+```shell
+# ./tools/minify/minify --config http://myserver/telegraf.conf
+```
+
+will download the configuration from `myserver`.
+
+The `--config` and `--config-dir` option can be used multiple times.
+If you want to build a Telegraf for multiple different environments,
+just specify the superset of all configuration files and directories.
+
+Building a minified telegraf involves the following steps
+
+1. [Configure](#configure-telegraf) Telegraf to use only the plugins
+   specified using the config files.
+1. [Build](#build-telegraf) the Telegraf binary
+1. [Restore](#restore-telegraf) the initial Telegraf state of including all plugins
+
+The `minify` command above will run all those steps for you. However,
+to have more control over the individual steps, `minify` provides
+_commands_ for the individual steps of the process. The steps and the
+associated commands are briefly described below. To get more help run
+
+```shell
+# ./tools/minify/minify --help
+```
+
+or
+
+```shell
+# ./tools/minify/minify <command> --help
+```
+
+### Configure Telegraf
+
+The core functionality is the configuration of Telegraf. This can be
+done using the `configure` command. To include all plugins in the
+Telegraf binary use
+
+```shell
+# ./tools/minify/minify configure --all-yes
+```
+
+This command will enable all plugins. To reduce the binary size and
+only configure required plugins you need to specify one or more
+configuration file(s) or one or more configuration directories.
+
+```shell
+# ./tools/minify/minify configure              \
+    --config     /etc/telegraf/telegraf.conf   \
+    --config-dir /etc/telegraf/telegraf.d
+```
+
+To get more output and see which plugins are enabled, you can specify
+the `--verbose` flag. You can also simply print the plugins that would
+be enabled without actually doing anything using
+
+```shell
+# ./tools/minify/minify configure          \
+    --config /etc/telegraf/telegraf.conf   \
+    --config-dir /etc/telegraf/telegraf.d  \
+    --dry-run                              \
+    --verbose
+```
+
+### Build Telegraf
+
+The `build` command will build the currently configured state
+of Telegraf via
+
+```shell
+# ./tools/minify/minify build
+```
+
+Alternatively, you can build Telegraf in a standard way using
+
+```shell
+# make clean all
+```
```

Comment:
I was looking through his more and wondered why this subcommand exists. I think if a new user jumps in, it is going to be very confusing as to why the `build` command doesn't actually respect a `--config` parameter, given the purpose of this tool is to build a smaller telegraf.

Reply:
Well a new user does not need to know. I need this step anyway, so I thought exposing it might not be an issue... Do you want me to remove it?

Reply:
Yeah, I think both the `configure` and `build` subcommand need not be exposed. Keep it simple and to only the building with a config/config-dir.

Reply:
It does _not_ simplify the code. By no means, but it takes away the possibility for

1. Resetting to the initial state (all plugins included) without triggering a build
2. More advanced usage such as manually disabling serializers etc.

Why should we remove this if it does not simplify anything???

Reply:
> It does not simplify the code.

My concern is not the code. This concern was with what is exposed to the user and their interaction with the CLI subcommand. You are now providing a `build` command that doesn't relate to the purpose of minifying in any way. It does not belong.

> Resetting to the initial state (all plugins included) without triggering a build

Why would the state be different? Again, as I said in another part, this is a git repo. They can use git commands to reset files if something goes wrong. You keep bringing up this scenario, and I do not see it as large of an issue as you do.

> More advanced usage such as manually disabling serializers etc.

You are going out of scope again. If someone is manually disabling things, then they should not be using minify.

Reply:
I've hidden the build command and added an example on default usage. Does that mitigate your concerns?